tests/fixtures/endpoints_handler.py

```diff
@@ -0,0 +1,206 @@
+    Tag endpoint functions with EndpointHandler.get("/some/url-with/{variables}").
+    .post and .patch are also implemented at the moment.
+    The regex compiler function will turn the url specified in the decorator function
+    into a regex string capable of capturing the variables in the url (curly brackets)
+    with named groups. That in turn enables linking the named path variables to the
+    variables in the endpoint function itself.
+    """
+
+    methods: dict[str, list[MatchableEndpoint]] = {
+        "GET": [],
+        "POST": [],
+        "PATCH": [],
+    }
+
+    @staticmethod
+    def compile_regex_url(url_pattern: str):
+        """Given a url pattern, compile a regex that matches named groups where specified
+        e.g. "/work-packages/{package_id}" would become "/work-packages/(?P<package_id>[^\/]+)"
+        And when a request URL like /work-packages/12 is matched against the regex-url above,
+        the match object will have a .groupdict() of {"package_id": "12"}
+        """
+
+        strip = "{}"
+        parameter_pattern = re.compile(r"{.*?}")
+
+        url = re.sub(
+            parameter_pattern,
+            repl=lambda name: f"(?P<{name.group().strip(strip)}>[^/]+)",
+            string=url_pattern,
+        )
+        return url
+
+    @classmethod
+    def _add_endpoint(cls, method: str, url: str, endpoint_function: Callable):
+        """Process url and store endpoint according to method type"""
+        url_pattern = EndpointsHandler.compile_regex_url(url)
+        matchable_endpoint = MatchableEndpoint(
+            url_pattern=url_pattern,
+            method=method,
+            endpoint_function=endpoint_function,
+        )
+        cls.methods[method].append(matchable_endpoint)
+        cls.methods[method].sort(
+            key=lambda endpoint: len(endpoint.url_pattern), reverse=True
+        )
+
+    @classmethod
+    def get(cls, url: str):
+        """Decorator function to add endpoint to Handler"""
+
+        def inner(endpoint_function: Callable):
+            cls._add_endpoint(
+                method="GET", url=url, endpoint_function=endpoint_function
+            )
+            return endpoint_function
+
+        return inner
+
+    @classmethod
+    def post(cls, url: str):
+        """Decorator function to add endpoint to Handler"""
+
+        def inner(endpoint_function: Callable):
+            cls._add_endpoint(
+                method="POST", url=url, endpoint_function=endpoint_function
+            )
+            return endpoint_function
+
+        return inner
+
+    @classmethod
+    def patch(cls, url: str):
+        """Decorator function to add endpoint to Handler"""
+
+        def inner(endpoint_function: Callable):
+            cls._add_endpoint(
+                method="PATCH", url=url, endpoint_function=endpoint_function
+            )
+            return endpoint_function
+
+        return inner
+
+    @staticmethod
+    def convert_parameter_types(
```

Comment:
```suggestion
    def _convert_parameter_types(
```
Only used from within the class, as far as I can see

Reply:
Changed in [f9a5cf1](https://github.com/ghga-de/ghga-connector/pull/68/commits/f9a5cf199803a93f4f8f26826410e0c883a01d7b)